crosspoint_reader/converter.py

```
@@ -0,0 +1,509 @@
+                        
+                        # Replace block patterns (p/div with span and img)
+                        block_pattern = re.compile(
+                            r'(<(?:p|div)[^>]*>\s*<span>\s*<img[^>]*src=["\'][^"\']*(?:' + 
+                            re.escape(orig_name) + '|' + re.escape(new_name) + 
+                            r')[^>]*/?>\s*</span>\s*</(?:p|div)>)',
+                            re.IGNORECASE | re.DOTALL
+                        )
+                        
+                        def replace_block(match):
+                            result = []
+                            for i, part in enumerate(parts):
```

Ruff (0.15.2)

[warning] Function definition does not bind loop variable `parts`

(B023)